src/mappings/utils.ts

```diff
@@ -288,6 +309,61 @@ export async function updateTotalDelegation(
   await delegator.save();
 }
 
+export async function updateIndexerCapacity(
+  address: string,
+  event: FrontierEvmEvent
+): Promise<void> {
+  const indexer = await Indexer.get(address);
+  const delegationId = getDelegationId(address, address);
+  const delegation = await Delegation.get(delegationId);
+  const staking = Staking__factory.connect(
+    STAKING_ADDRESS,
+    new FrontierEthProvider()
+  );
+  const eraManager = EraManager__factory.connect(
+    ERA_MANAGER_ADDRESS,
+    new FrontierEthProvider()
+  );
+
+  const leverageLimit = await staking.indexerLeverageLimit();
+
+  if (indexer) {
+    const indexerStake = delegation?.amount;
+    const indexerTotalStake = indexer?.totalStake;
+
+    const stakeCurr = BigNumber.from(indexerStake?.value.value ?? 0);
+    const stakeAfter = BigNumber.from(indexerStake?.valueAfter.value ?? 0);
+
+    const totalStakeCurr = BigNumber.from(indexerTotalStake?.value.value ?? 0);
+    const totalStakeAfter = BigNumber.from(
+      indexerTotalStake?.valueAfter.value ?? 0
+    );
+
+    const current =
+      stakeCurr?.mul(leverageLimit).sub(totalStakeCurr || 0) ||
```

Comment:
as you can have default value for `stakeCurr` etc, doesn't need to have `?.` and other default values in this expression.

Same with after